core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -197,14 +207,7 @@ class ZkUtils(val zkClient: ZkClient,
                               BrokerSequenceIdPath,
                               IsrChangeNotificationPath)
 
-  val securePersistentZkPaths = Seq(BrokerIdsPath,
```

Comment:
Note that ZK ACL is not recursive. So, if you set permission at /parent, that permission doesn't apply at /parent/child. That's why we have to explicitly set the permission at the lowest directory level. So, could you just add the missing ZK paths here and add a comment above that future new persistent ZK paths may need to be added to this list? It would be useful to keep using the named constant to avoid typos. In addition to the ACL path, the following paths are also missing.

getEntityConfigRootPath(ConfigType.User)
getEntityConfigRootPath(ConfigType.Broker)